builtin/providers/aws/resource_aws_efs_mount_target_test.go

```diff
@@ -57,6 +57,21 @@ func TestAccAWSEFSMountTarget_basic(t *testing.T) {
 	})
 }
 
+func TestResourceAWSEFSMountTarget_dnsName(t *testing.T) {
+	// Seed values.
+	az := "non-existant-1c"
+	region := "non-existant-1"
+	id := "fs-123456ab"
+
+	expected := fmt.Sprintf("%s.%s.efs.%s.amazonaws.com", az, id, region)
```

Comment:
I think that a black-box approach here would be more useful. i.e. The function should not try to cover all possible cases, because that would mean we'd have to reimplement the function again inside a test - which is what you did here.

e.g. typically you'd be testing that
```
testedSum(1, 1) == 2
testedSum(3, 4) == 7
```
rather than
```
mySumInTest(a, b) == testedSum(a, b)
```

Having static and more stable tests also makes it easier to spot a bug.